packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -179,16 +191,50 @@ class CupertinoSwitch extends StatefulWidget {
   /// See also:
   ///
   ///  * [trackColor], the color to use for the track when the switch is off.
-  final Color? activeColor;
+  Color? get activeColor => _activeColor;
```

Comment:
```suggestion
  Color? get activeColor => activeTrackColor ?? _activeColor;
```
This will make tests backwards compatible. As well as lowering the chances that packages or other outside code that look at this value are able to get something appropriate.